src/trio/_tests/test_timeouts.py

```diff
@@ -75,6 +75,66 @@ async def sleep_3() -> None:
     await check_takes_about(sleep_3, TARGET)
 
 
+@slow
+async def test_move_on_after_shields_from_outer() -> None:
+    duration = 0.1
```

Comment:
Given that this is using sleep a checkpoint, can you use a checkpoint (or trio.sleep(0)) instead of a hardcoded small value?

Reply:
If i use 0, i don't think i can make use of `check_takes_about` because it checks `dur / expected_dur < 1.5`, and `expected_dur = 0` in this case. The docstring also says that they observe overruns when they sleep for 0.05s, so I picked 0.1s here.
Alternatively, I can use a different way to test whether we actually sleep (instead of using `check_takes_about`)

Reply:
So don't use `check_takes_about` (and don't mark this test as slow). Instead do something like:
```
try:
    await trio.lowlevel.checkpoint()
except trio.Cancelled:
    pytest.fail("shield didn't work")
inner.shield = False
with pytest.raises(trio.Cancelled):
    await trio.lowlevel.checkpoint()
```